azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/implementation/AvailabilitySetImpl.java

```diff
@@ -101,28 +101,32 @@ public ServiceCall applyAsync(ServiceCallback<AvailabilitySet> callback) {
         return this.createAsync(callback);
     }
 
+    // CreatorTaskGroup.ResourceCreator implementation
+
     @Override
-    protected void createResource() throws Exception {
+    public Resource createResource() throws Exception {
```

Comment:
Btw I tried to explain the changes in this PR in the PR description, i can expand the description if anything is unclear.